packages/artifact/src/internal-utils.ts

```diff
@@ -96,20 +119,61 @@ export function getArtifactUrl(): string {
  * from the server if attempted to be sent over. These characters are not allowed due to limitations with certain
  * file systems such as NTFS. To maintain platform-agnostic behavior, all characters that are not supported by an
  * individual filesystem/platform will not be supported on all fileSystems/platforms
+ *
+ * FilePaths can include characters such as \ and / which are not permitted in the artifact name alone
  */
-const invalidCharacters = ['\\', '/', '"', ':', '<', '>', '|', '*', '?', ' ']
+const invalidArtifactFilePathCharacters = [
+  '"',
+  ':',
+  '<',
+  '>',
+  '|',
+  '*',
+  '?',
+  ' '
+]
+const invalidArtifactNameCharacters = [
+  '"',
+  ':',
+  '<',
+  '>',
+  '|',
+  '*',
+  '?',
+  ' ',
+  '\\',
+  '/'
+]
```

Comment:
I think this would be more clear if `invalidArtifactFilePathCharacters` was explicitly a subset of `invalidArtifactNameCharacters`. Something like:
```suggestion
const invalidArtifactNameCharacters = [...invalidArtifactFilePathCharacters, '\\', '/']
```